data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/PeerForwarder.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.peerforwarder;
+
+import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.peerforwarder.discovery.StaticPeerListProvider;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.linecorp.armeria.client.WebClient;
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class PeerForwarder {
+    private static final Logger LOG = LoggerFactory.getLogger(PeerForwarder.class);
+    private static final WebClient LOCAL_CLIENT = null;
+    public static final int ASYNC_REQUEST_THREAD_COUNT = 200;
+
+    private final PeerForwarderClientFactory peerForwarderClientFactory;
+    private final ObjectMapper objectMapper;
+    private HashRing hashRing;
+    private PeerClientPool peerClientPool;
+    private final ExecutorService executorService;
+
+    public PeerForwarder(final PeerForwarderClientFactory peerForwarderClientFactory,
+                         final ObjectMapper objectMapper) {
+        this.peerForwarderClientFactory = peerForwarderClientFactory;
+        this.objectMapper = objectMapper;
+
+        executorService = Executors.newFixedThreadPool(ASYNC_REQUEST_THREAD_COUNT);
+    }
+
+    public List<Record<Event>> processRecords(final Collection<Record<Event>> records, final Set<String> identificationKeys) {
+        // TODO: decide the default behaviour of Core Peer Forwarder and move the HashRing and PeerClientPool creation to constructor
+        hashRing = peerForwarderClientFactory.createHashRing();
+        peerClientPool = peerForwarderClientFactory.setPeerClientPool();
```

Comment:
I think both of them can still be injected in the constructor.

Reply:
Explained above why it's not in the constructor

Reply:
Do we have to create new instance everytime forwardRecords is called? Will move L44-45 into constructor break things?